packages/flutter/lib/src/material/page.dart

```diff
@@ -127,8 +129,76 @@ mixin MaterialRouteTransitionMixin<T> on PageRoute<T> {
 
   @override
   Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
-    final PageTransitionsTheme theme = Theme.of(context).pageTransitionsTheme;
-    return theme.buildTransitions<T>(this, context, animation, secondaryAnimation, child);
+    final ThemeData themeData = Theme.of(context);
+    return ValueListenableBuilder<bool>(
+        valueListenable: navigator?.userGestureInProgressNotifier ?? ValueNotifier<bool>(false),
```

Comment:
I think we can assume navigator will always be non null

Reply:
Ok, I replaced `?.` with `!.`